wave_kinematics/tests/rotation_test.cpp

```diff
@@ -0,0 +1,185 @@
+    // rotation1_inverse is the inverse of rotation1.
+    wave::Rotation::fromEulerXYZ(-0.037879880513201,
+                                 -0.220124031212965,
+                                 -0.285771700628461,
+                                 rotation1_inverse);
+    eye3.rotation_object.setMatrix(1, 0, 0, 0, 1, 0, 0, 0, 1);
+
+    // rotation1_true is set using the underlying element wise initializer.
+    // It is the expected rotation matrix, using the parameters from
+    // rotation1.
+
+    rotation1_truth.rotation_object.setMatrix(0.936293363584199,
+                                              -0.275095847318244,
+                                              0.218350663146334,
+                                              0.289629477625516,
+                                              0.956425085849232,
+                                              -0.036957013524625,
+                                              -0.198669330795061,
+                                              0.097843395007256,
+                                              0.975170327201816);
+
+
+    v1 << 1, 1, 1;
+    v1_rotated << 0.879548179412290, 1.209097549950123, 0.874344391414010;
+    rotation1_angleaxis_params << 0.068924613882066, 0.213225926957886,
+      0.288748939228675;
+  }
+};
+
+
+// Construction tests.
+
+// Test that fromEulerXYZ returns the correct matrix.
+TEST_F(RotationTestFixture, test_from_euler_xyz) {
+  ASSERT_TRUE(rotation1_truth.isNear(rotation1));
+}
+
+
+// Test that fromAngleAxis returns the correct matrix.
+TEST_F(RotationTestFixture, test_from_angle_axis) {
+  wave::Rotation rotation_from_angle_axis;
+  wave::Rotation::fromAngleAxis(rotation1_angleaxis_params.norm(),
+                                rotation1_angleaxis_params.normalized(),
+                                rotation_from_angle_axis);
+  ASSERT_TRUE(rotation1.isNear(rotation_from_angle_axis));
+}
+// Test that fromExpMap returns the correct matrix.
+TEST_F(RotationTestFixture, test_from_exp_map) {
+  wave::Rotation rotation_from_exp_map;
+  wave::Rotation::fromExpMap(rotation1_angleaxis_params,
+                             rotation_from_exp_map);
+  ASSERT_TRUE(rotation1.isNear(rotation_from_exp_map));
+}
+
+// Test that logMap returns the correct parameters.
+TEST_F(RotationTestFixture, test_log_map) {
+  wave::Vec3 se3_params = rotation1.logMap();
+  wave::Vec3 error = rotation1_angleaxis_params - se3_params;
+  ASSERT_LE(error.norm(), wave::Rotation::comparision_threshold);
+}
+
+
+// Test to ensure default constructor produces identity.
+TEST_F(RotationTestFixture, test_default_constructor) {
+  wave::Rotation R;
+  ASSERT_TRUE(R.isNear(eye3));
+}
+
+
+// Test to ensure setToIdentity returns identity.
+
+TEST_F(RotationTestFixture, test_set_to_identity) {
+  // First make sure the rotation is not already identity.
+  ASSERT_FALSE(rotation1.isNear(eye3));
+  rotation1.setToIdentity();
+  ASSERT_TRUE(rotation1.isNear(eye3));
```

Comment:
For my tests I use a helper predicate like this: http://stackoverflow.com/a/39238772/431033

The benefit is if it fails, gtest will print both matrices.

(Except that probably won't work here, because the Rotation type has no printing method defined - another reason not to wrap it)